core/controllers/contributor_dashboard_admin.py

```diff
@@ -307,33 +398,74 @@ def _get_complete_translation_contribution_stats(
                     months of format: "%b %Y", e.g. "Jan 2021".
             Unnecessary keys language_code, topic_id, contribution_dates,
             contributor_user_id are consequently deleted.
+
+        Raises:
+            Exception. No topic associated with the
+                TranslationContributionStats.
         """
         translation_contribution_stats_dicts = [
             stats.to_dict() for stats in translation_contribution_stats
         ]
-        topic_ids = [
-            stats_dict['topic_id']
-            for stats_dict in translation_contribution_stats_dicts
-        ]
+        topic_ids: List[str] = []
+        for stats_dict in translation_contribution_stats_dicts:
+            if stats_dict['topic_id'] is None:
+                raise Exception(
+                    'No topic associated with the TranslationContributionStats.'
+                )
+            topic_ids.append(stats_dict['topic_id'])
         topic_summaries = topic_fetchers.get_multi_topic_summaries(topic_ids)
         topic_name_by_topic_id = {}
         for topic_summary in topic_summaries:
             if topic_summary is None:
                 continue
             topic_name_by_topic_id[topic_summary.id] = topic_summary.name
+
+        response_translation_contribution_stats_dicts: (
+            List[TranslationContributionStatsDict]
+        ) = []
         for stats_dict in translation_contribution_stats_dicts:
-            stats_dict['topic_name'] = topic_name_by_topic_id.get(
-                stats_dict['topic_id'], 'UNKNOWN')
-            stats_dict['contribution_months'] = list({
-                contribution_date.strftime('%b %Y')
-                for contribution_date in stats_dict['contribution_dates']
+            assert stats_dict['topic_id'] is not None
+            if stats_dict['language_code'] is None:
+                raise Exception(
+                    'No language_code exists for the give Topic having '
```

Comment:
`No language_code exists for the topic with topic_id %s`

That said, I'm a bit confused -- does the language code not exist for the topic, or in the stats dict? These are different things.

Reply:
> does the language code not exist for the topic, or in the stats dict? 

This exception is for the stats dict, and the error statement is changed for the same.